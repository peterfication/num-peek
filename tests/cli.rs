use std::path::Path;

use assert_cmd::Command;
use insta::assert_snapshot;

#[test]
fn cli_help_output() {
    let mut cmd = Command::cargo_bin("num-peek").unwrap();
    let result = cmd.arg("--help").assert().success();

    let output = String::from_utf8(result.get_output().stdout.clone()).unwrap();

    assert_snapshot!(output);
}

fn cli_run_with_file(file_path: &str, snapshot_name: &str) {
    let mut cmd = Command::cargo_bin("num-peek").unwrap();
    let result = cmd.arg(file_path).assert().success();
    let output = String::from_utf8(result.get_output().stdout.clone()).unwrap();

    assert_snapshot!(snapshot_name, output);
}

#[test]
fn cli_run_all_types() {
    let files = [
        "assets/bool.npy",
        "assets/uint8.npy",
        "assets/uint16.npy",
        "assets/uint32.npy",
        "assets/uint64.npy",
        "assets/int8.npy",
        "assets/int16.npy",
        "assets/int32.npy",
        "assets/int64.npy",
        "assets/float16.npy",
        "assets/float32.npy",
        "assets/float64.npy",
    ];
    for file in files {
        let snapshot_name = format!(
            "{}_snapshot",
            Path::new(file).file_stem().unwrap().to_string_lossy()
        );
        cli_run_with_file(file, &snapshot_name);
    }
}
