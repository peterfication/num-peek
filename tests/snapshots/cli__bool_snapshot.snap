---
source: tests/cli.rs
expression: output
---
Peek into assets/bool.npy
----------------------------------------
Dimensions: 2
Shape: [2, 3]
Type: Bool8
Bytes: 134 B
----------------------------------------
Number of values: 6
Number of unique values: 2
Unique values: [true, false]
